src/cheri_insts.sail

```diff
@@ -1370,6 +1370,66 @@ function clause execute (LoadCapCap(cd, cs1)) = {
   handle_load_cap_via_cap(cd, 0b0 @ cs1, cs1_val, vaddr)
 }
 
+union clause ast = CLoadTags : (regidx, regidx)
+function clause execute (CLoadTags(rd, cs1)) = {
+  let cs1_val = C(cs1);
+  if not (cs1_val.tag) then {
+    handle_cheri_reg_exception(CapEx_TagViolation, cs1);
+    RETIRE_FAIL
+  } else if isCapSealed(cs1_val) then {
+    handle_cheri_reg_exception(CapEx_SealViolation, cs1);
+    RETIRE_FAIL
+  } else if not (cs1_val.permit_load) then {
+    handle_cheri_reg_exception(CapEx_PermitLoadViolation, cs1);
+    RETIRE_FAIL
+  } else if not (cs1_val.permit_load_cap) then {
+    handle_cheri_reg_exception(CapEx_PermitLoadCapViolation, cs1);
+    RETIRE_FAIL
+  } else {
+    let vaddr = cs1_val.address;
```

Comment:
This can be hoisted to avoid indenting (as is effectively done with the normal loads/stores due to calling the helpers with the desired vaddr).